addons/mail/static/src/models/autocomplete-input-view/autocomplete-input-view.js

```diff
@@ -0,0 +1,294 @@
+                    } else {
+                        this.discuss.handleAddChatAutocompleteSource(req, res);
+                    }
+                };
+            }
+            if (this.discussSidebarCategoryAsItemNewInput) {
+                return this.discussSidebarCategoryAsItemNewInput.onAddItemAutocompleteSource;
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                /**
+                 * @param {Object} req
+                 * @param {string} req.term
+                 * @param {function} res
+                 */
+                return (req, res) => {
+                    const value = _.escape(req.term);
+                    this.messaging.models['Partner'].imSearch({
+                        callback: partners => {
+                            const suggestions = partners.map(partner => {
+                                return {
+                                    id: partner.id,
+                                    value: partner.nameOrDisplayName,
+                                    label: partner.nameOrDisplayName,
+                                };
+                            });
+                            res(_.sortBy(suggestions, 'label'));
+                        },
+                        keyword: value,
+                        limit: 10,
+                    });
+                };
+            }
+            return clear();
+        },
+    },
+    fields: {
+        chatWindowOwnerAsNewMessageFormInput: one2one('ChatWindow', {
+            inverse: 'newMessageFormInputView',
+            readonly: true,
+        }),
+        /**
+         * States the OWL component of this autocomplete input view.
+         */
+        component: attr(),
+        customClass: attr({
+            compute: '_computeCustomClass',
+            default: '',
+        }),
+        discussAsMobileAddItemHeader: one2one('Discuss', {
+            inverse: 'mobileAddItemHeaderInputView',
+            readonly: true,
+        }),
+        discussSidebarCategoryAsItemNewInput: one2one('DiscussSidebarCategory', {
+            inverse: 'itemNewInputView',
+            readonly: true,
+        }),
+        focus: attr({
+            default: () => {},
+        }),
+        isFocusOnMount: attr({
+            compute: '_isFocusOnMount',
+            default: false,
+        }),
+        isHtml: attr({
+            compute: '_computeIsHtml',
+            default: false,
+        }),
+        messagingMenuAsMobileNewMessageInput: one2one('MessagingMenu', {
+            inverse: 'mobileNewMessageInputView',
+            readonly: true,
+        }),
+        placeholder: attr({
+            compute: '_computePlaceholder',
+            default: '',
+        }),
+        select: attr({
+            compute: '_computeSelect',
+            default: (ev, ui) => {},
+        }),
+        source: attr({
+            compute: '_computeSource',
+            default: (req, res) => {},
+        }),
```

Comment:
Why have compute that return functions?
Can't we just create one handler function in the model for each case, and have the conditions on owner inside the handler?

Reply:
It's been done that way to minimise logic in `AutocompleteInputComponent`, and the passing of function with these params are part of integration with jQuery `ui-autocomplete` that we still rely on for autocomplete feature.

Reply:
This doesn't answer my comment, I know we need a function but we should only have one handle function, with conditions inside, instead of conditionally defining the function.